test/source/directionality.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+// Spack Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+#include <doctest/doctest.h>
+#include "smeagle/smeagle.h"
```

Comment:
I really love this setup! It will be really easy to write functions and matching tests.